core/src/main/java/org/jruby/ext/bigdecimal/RubyBigDecimal.java

```diff
@@ -1040,7 +1035,9 @@ public IRubyObject op_quo(ThreadContext context, IRubyObject other) {
         int pow = len / 4;
         int precision = (pow + 1) * 4 * 2;
 
-        return divWithScale(context, val, precision);
+        IRubyObject result = divWithScale(context, val, precision);
+        if (result.getClass() == RubyBigDecimal.class) return ((RubyBigDecimal) result).setResult();
```

Comment:
why not an `instanceof` check?

Reply:
Paranoia that there might be some hidden RubyBigDecimal subclass with strange behaviour. 

Nah, it can totally be changed to `instanceof`.

Reply:
thx, if you get a chance please adjust - JRuby is quite in control of its sub-classes. 
no need for paranoia :wink:

Reply:
Sure, I pushed a fix.